gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -1837,7 +1836,24 @@ bool Highlightable::TryUnlock(Actor *actor, bool removekey) const {
 
 	if (removekey) {
 		CREItem *item = NULL;
-		haskey->inventory.RemoveItem(KeyResRef,0,&item);
+		int result = haskey->inventory.RemoveItem(KeyResRef,0,&item);
+		if (result == -1) {
+			int i= haskey->inventory.GetSlotCount();
+			while (i--) {
+				//maybe we could speed this up if we mark bag items with a flags bit
+				const CREItem *itemslot = haskey->inventory.GetSlotItem(i);
+				if (!itemslot)
+					continue;
+				const Item *itemStore = gamedata->GetItem(itemslot->ItemResRef);
+				if (!itemStore)
+					continue;
+				if (core->CanUseItemType(SLOT_BAG, itemStore)) {
```

Comment:
This function does a lot of things and its use here is not so obvious. Do you need more than the `int ret = (slotmatrix[item->ItemType]&slottype);` check?

Reply:
I only need to now if is a SLOT_BAG, how can I get the slotmatrix ?

Reply:
You'll then need to add an accessor to Interface. Maybe `CheckItemType(Item*, int slotType)`. I see it can then be used elsewhere as well.

Reply:
Added you method in Interface.